ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java

```diff
@@ -309,6 +310,34 @@ public APIGatewayProxyResponseEvent handleRequest(
 
             var userIdentityError = validateUserIdentityResponse(userIdentityUserInfo);
             if (userIdentityError.isPresent()) {
+                var accountInterventionStatus =
+                        getAccountInterventionStatus(pairwiseSubject.getValue());
+                if (configurationService.isAccountInterventionServiceEnabled()) {
+                    if (accountInterventionStatus.blocked()) {
+                        // TODO: back channel logout + redirect to blocked page
+                        LOG.info("Account is blocked");
+                    } else if (accountInterventionStatus.suspended()
+                            || accountInterventionStatus.resetPassword()
+                            || accountInterventionStatus.reproveIdentity()) {
+                        // TODO: back channel logout + redirect to suspended page
+                        LOG.info(
+                                "Account is suspended, requires a password reset, or requires identity to be reproved");
+                    }
+                }
+
+                var returnCode = userIdentityUserInfo.getClaim(RETURN_CODE.getValue());
+                if (returnCode instanceof List<?> && !((List<?>) returnCode).isEmpty()) {
```

Comment:
Is there any reason to cast to `List<?>` in the second condition if the first condition already ensures that `returnCode` is a `List<?>`?

Reply:
`returnCode` is still of type `Object` (even though we know it is a `List`) which means we cannot access operations specific to `List`, such as `isEmpty()`, hence the need to cast it